### PR TITLE
MMT-3529: As a user, I can view all collection permissions

### PR DESCRIPTION
**Pull request by dmistry1**
# Overview

### What is the feature?
Implements the ability to mange collection permission 

# Checklist

- [X] I have added automated tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings

### CODE REVIEW

File: static/src/js/App.jsx [hunk 1]
```diff
@@ -87,7 +88,7 @@ export const App = () => {
       const { items } = acls
 
       if (items.length > 0) {
-        const providerList = items.map(({ acl }) => acl.provider_identity.provider_id)
+        const providerList = items.map((item) => item.providerIdentity.provider_id)
```
Comment (abbottry) [inline static/src/js/App.jsx:91]: @dmistry1 is this still going to be snake cased? Your GQL PR calls camelcaseKeys in a few spots.

File: static/src/js/components/Permission/Permission.jsx [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+  const granuleConstraintSummary = () => {
+    const {
+      accessValue: granuleAccessValue,
+      temporal: granuleTemporal = null
+    } = granuleIdentifier || {}
+
+    const sentenceFragments = ['This permission']
+
+    if (granuleApplicable) {
+      sentenceFragments.push(' grants its assigned groups access to granules')
```
Comment (abbottry) [inline static/src/js/components/Permission/Permission.jsx:170]: Unless there is some kind of punctuation you're dealing with, might be best to add these strings without leading spaces and join them all at the end with `' '`

File: static/src/js/components/PermissionList/PermissionList.jsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                    {
+                      totalPages > 1 && (
+                        <Col xs="auto">
+                          {pagination}
+                        </Col>
+                      )
+                    }
+                  </Row>
+                  <Table
+                    id="permission-list-table"
```
Comment (abbottry) [inline static/src/js/components/PermissionList/PermissionList.jsx:113]: Trying to keep props sorted

File: static/src/js/components/permissionCollectionTable/PermissionCollectionTable.jsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  const buildEllipsisLinkCell = useCallback((cellData, rowData) => {
+    const { conceptId: collectionConceptId } = rowData
+
+    return (
+      <EllipsisLink to={`/collections/${collectionConceptId}`}>
+        {cellData}
+      </EllipsisLink>
+    )
+  }, [])
+  const collectionColumns = [
```
Comment (abbottry) [inline static/src/js/components/permissionCollectionTable/PermissionCollectionTable.jsx:69]: Add a new line.

File: static/src/js/components/permissionCollectionTable/PermissionCollectionTable.jsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    }
+  ]
+
+  return (
+    <Row>
+      <Col md={12}>
+        {
+          items && (
+            <Table
+              id="permission-collection-table"
```
Comment (abbottry) [inline static/src/js/components/permissionCollectionTable/PermissionCollectionTable.jsx:97]: Trying to keep props sorted.

File: static/src/js/operations/queries/getAvailableProviders.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { gql } from '@apollo/client'
+
+// Query to available providers acls
+export const GET_AVAILABLE_PROVIDERS = gql`
+  query AvailableProviders($params: AclsInput) {
```
Comment (abbottry) [inline static/src/js/operations/queries/getAvailableProviders.js:5]: Name the query to match its action, this one should be `GetAvailableProviders`

File: static/src/js/operations/queries/getCollectionPermissions.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { gql } from '@apollo/client'
+
+export const GET_COLLECTION_PERMISSIONS = gql`
+  query Acls($params: AclsInput) {
```
Comment (abbottry) [inline static/src/js/operations/queries/getCollectionPermissions.js:4]: Rename this query `GetCollectionPermissions`.

File: static/src/js/pages/PermissionListPage/PermissionListPage.jsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    breadcrumbs={
+      [
+        {
+          label: 'Collection Permissions',
+          active: true
+        }
+      ]
+    }
+    pageType="secondary"
+    title="Permissions"
```
Comment (abbottry) [inline static/src/js/pages/PermissionListPage/PermissionListPage.jsx:28]: Should this be `Collection Permissions`?

File: static/src/js/pages/PermissionPage/PermissionPage.jsx [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+  const { acl } = data
+  const { name } = acl
+
+  return (
+    <PageHeader
+      breadcrumbs={
+        [
+          {
+            label: 'Collection Permission',
```
Comment (abbottry) [inline static/src/js/pages/PermissionPage/PermissionPage.jsx:41]: This should be plural.

**Review by abbottry — CHANGES_REQUESTED** [review]

File: static/src/js/App.jsx (hunk 1 above)
Comment (dmistry1) [reply static/src/js/App.jsx:91]: Yes this provider is still going to be snakedCase, the camelcaseKeys I has was only for groups and catalogIdentity

Printed out the response from graphql to verify.
image

File: static/src/js/components/Permission/__tests__/Permission.test.jsx [hunk 10]
```diff
@@ -383,13 +383,13 @@ describe('Permission', () => {
...
 
-      expect(screen.getByText('This permission grants its assigned groups access to all of its collections that have access constraint values between 1 and 100000 (or are undefined) and that have a start and end date having some overlap with the date range 2018-04-01T04:07:58Z to 2023-03-29T04:11:01Z')).toBeInTheDocument()
-      expect(screen.getByText('This permission grants its assigned groups access to granules that have access constraint values between 1 and 10000 (or are undefined) and that have a start and end date having some overlap with the date range 2018-04-01T04:07:58Z to 2023-03-29T04:11:01Z that belong to any of its collections that have access constraint values between 1 and 100000 (or are undefined) and that have a start and end date having some overlap with the date range 2018-04-01T04:07:58Z to 2023-03-29T04:11:01Z')).toBeInTheDocument()
+      expect(screen.getByText('This permission grants its assigned groups access to all of its collections that have access constraint values between 1 and 100000 (or are undefined), and, that have a start and end date having some overlap with the date range 2018-04-01T04:07:58Z to 2023-03-29T04:11:01Z')).toBeInTheDocument()
```
Comment (abbottry) [inline static/src/js/components/Permission/__tests__/Permission.test.jsx:386]: This is what I meant by punctuation, you may want to revert these changes because this result is worse than the previous. Alternatively you can figure out this `, and,`.

**Review by abbottry — CHANGES_REQUESTED** [review]

**Review by abbottry — APPROVED** [review]
